server/xpub-model/entities/manuscript/index.test.js

```diff
@@ -140,6 +141,117 @@ describe('Manuscript', () => {
     })
   })
 
+  describe('get fileStatus()', () => {
+    describe('given there are no files', () => {
+      it('returns READY', () => {
+        const manuscript = new Manuscript({
+          createdBy: userId,
+        })
+        manuscript.files = []
+        expect(manuscript.fileStatus).toEqual('READY')
+      })
+    })
+
+    describe('given there is a single file', () => {
+      let manuscript
+      let file
+      let setStatusOfFirstFile
+
+      beforeEach(async () => {
+        manuscript = await createInitialManuscript(userId)
+        manuscript = await addFileToManuscript(manuscript)
+        file = await File.find(manuscript.files[0].id)
+        setStatusOfFirstFile = setStatusOfFile.bind(null,
+          file, manuscript
+        )
+      })
+
+      it('returns READY when the file is stored', async () => {
+        manuscript = await setStatusOfFirstFile('STORED')
+        expect(manuscript.fileStatus).toEqual('READY')
+      })
+
+      it('returns READY when the file upload was cancelled', async () => {
+        manuscript = await setStatusOfFirstFile('CANCELLED')
+        expect(manuscript.fileStatus).toEqual('READY')
+      })
+
+      it('returns CHANGING when the file has been uploaded to the app server', async () => {
+        manuscript = await setStatusOfFirstFile('UPLOADED')
+        expect(manuscript.fileStatus).toEqual('CHANGING')
+      })
+
+      it('returns CHANGING when the file has been created in the database', async () => {
+        manuscript = await setStatusOfFirstFile('CREATED')
+        expect(manuscript.fileStatus).toEqual('CHANGING')
+      })
+    })
+
+    describe('given there are multiple files', () => {
+      let manuscript
+      let file1, file2
+      let setStatusOfFirstFile, setStatusOfSecondFile
+
+      beforeEach(async () => {
+        manuscript = await createInitialManuscript(userId)
+        manuscript = await addFileToManuscript(manuscript)
+        manuscript = await addFileToManuscript(manuscript);
```

Comment:
I think it would be nice to move these test helpers to a better location so we can reuse them. We've got some similar helpers within `model.test.js` and within this file.

👍 if you want me to raise an issue for it

Reply:
I don't know how the upcoming re-architecture is going to effect our helpers, so may be you should create the issue but noting that it was created before the re-architecture work is done...

Reply:
https://github.com/elifesciences/elife-xpub/issues/1692